subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -2391,11 +2394,31 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
 	if (!lll->is_hdcd)
 #endif /* CONFIG_BT_PERIPHERAL */
 	{
-		/* Apply random delay in range [0..ULL_ADV_RANDOM_DELAY] */
-		random_delay = ticker_update_rand(adv, ULL_ADV_RANDOM_DELAY,
-						  0, 0, ticker_update_op_cb);
+		if (IS_ENABLED(CONFIG_BT_CTLR_JIT_SCHEDULING) ||
+		    (ticker_update_req == ticker_update_ack)) {
+			/* Ticker update requested */
+			ticker_update_req++;
+
+			/* Apply random delay in range [0..ULL_ADV_RANDOM_DELAY] */
+			random_delay = ticker_update_rand(adv, ULL_ADV_RANDOM_DELAY, 0U, 0U,
+							  ticker_update_op_cb);
+#if defined(CONFIG_BT_CTLR_JIT_SCHEDULING)
+			adv->delay = random_delay;
+#endif /* CONFIG_BT_CTLR_JIT_SCHEDULING */
+		} else {
+			random_delay = 0U;
+		}
 
 #if defined(CONFIG_BT_CTLR_ADV_EXT)
+		uint16_t event_counter_inc;
+
+		if (lazy == TICKER_LAZY_MUST_EXPIRE) {
+			lazy = 0U;
+			event_counter_inc = 0U;
+		} else {
+			event_counter_inc = (lazy + 1U);
+		}
```

Comment:
Isn't it enough to set lazy = 0 for lazy == TICKER_LAZY_MUST_EXPIRE? I would expect the event counter to need to be incremented by 1 in that case also

Reply:
i agree... i am discussing with the mesh team, as the fix in this PR seems to break some mesh tests in bsim CI.

Reply:
@Tronil, the skipped event is not counted so that the events returned to Host when the adv set terminates reflects correctly the number of event transmitted on air. With this conclusion, the behavior in this PR remains as it is currently upstream.

Please review and provide your approval.

Reply:
Ah - yeah carefully reading the spec we should actually keep track of the value of *completed* *transmitted* extended advertising events. So I guess we are actually not being entirely correct:
1. adv->event_counter should be increment once the adv_aux event is done, not in this ticker_cb
2. adv->event_counter should always be incremented by 1, lazy should not be used

It doesn't really have anything to do with this change though, so we should probably address that separately.

Reply:
Agreed... would you be able to create github issue referencing the spec text?